invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/subjects.html

```diff
@@ -15,14 +15,15 @@
   <h3 class="hidden">{{ _('Keywords') }}</h3>
   <ul class="ui horizontal list no-bullets subjects">
     {%- for subject in keywords %}
-      <li class="item">
-        <a href='/search?q=metadata.subjects.subject:"{{ subject.subject }}"'
-           class="subject"
-           title="{{ _('Search results for ') + subject.subject }}"
-        >
-          {{ subject.subject }}
-        </a>
-      </li>
+    <li class="item">
+      {%- set q = 'metadata.subjects.subject:"' + subject.subject + '"' %}
```

Comment:
Ah, you are making the point for a change I didn't include but wanted to: this `subject` variable here is not a "subject", it's a keyword. As such it doesn't have an id. So in our instance where I am overriding this template to bring the fix, I renamed things:

```jinja
{%- set keywords_and_subjects = record.metadata.subjects -%}

{% if keywords_and_subjects %}
  {%- set keywords = keywords_and_subjects|rejectattr("scheme")|list %}
  {% if keywords %}
  <h3 class="hidden">{{ _('Keywords') }}</h3>
  <ul class="ui horizontal list no-bullets subjects">
    {%- for keyword in keywords %}
      <li class="item">
        {%- set q = 'metadata.subjects.subject:"' + keyword.subject + '"' %}
        <a href="{{ url_for('invenio_search_ui.search', q=q) }}"
          class="subject"
          title="{{ _('Search results for ') + keyword.subject }}"
        >
          {{ keyword.subject }}
        </a>
      </li>
    {%- endfor %}
  </ul>
  {% endif %}
```

It's not that ideal anyway since the fundamental field is called `metadata.subjects` and has a `subject` subfield that really means `label`, but such is life.

If the above renaming to `keywords_and_subjects` helps, I can put it in :).

Reply:
Thanks for the explanation! I think it's ok like that, I just didn't remember the duality of a subject i.e free text plus vocabulary....